rfcs/20190829-tfx-container-component-execution.md

```diff
@@ -0,0 +1,555 @@
+# TFX Container Component Execution Proposal
+
+Status         | Proposed
+:------------- | :-------
+**Author(s)**  | Ajay Gopinathan (<a href="mailto:ajaygopinathan@google.com">ajaygopinathan@google.com</a>), Hongye Sun (<a href="mailto:hongyes@google.com">hongyes@google.com</a>), Makoto Uchida (<a href="mailto:muchida@google.com">muchida@google.com</a>), Ruoyu Liu (<a href="mailto:ruoyu@google.com">ruoyu@google.com</a>)
+**Sponsor(s)** | Konstantinos Katsiapis (katsiapis@google.com), Pavel Dournov (dournov@google.com), Ramesh Chandra (rameshuc@google.com)
+**Updated**    | 2019-08-29
+
+## Objective
+
+This RFC proposes an orchestrator agnostic way to reliably execute a user’s
+container in the TFX pipeline. The proposal can support:
+
+*   Running an arbitrary container in either local docker environment or remote
+    k8s cluster.
+*   Passing data into the container
+*   Passing output data from the container
+*   Capturing logs from the container
+*   Handling errors and retries
+*   Cancelling the container execution if the pipeline is terminated
+
+## Motivation
+
+Currently, in a TFX pipeline, there is no way to execute a generic container as
+one of its steps. Without this feature, users cannot bring their own containers
```

Comment:
> Currently, in a TFX pipeline, there is no way to execute a generic container as
one of its steps.
Currently, the execution of a generic container as a step in a TFX pipeline is not supported.